transports/http/src/main/java/org/mule/transport/http/HttpMessageProcessTemplate.java

```diff
@@ -504,7 +506,14 @@ private void sendFailureResponseToClient(MessagingException exception, int httpS
 
     private Map<String,String> getThrottlingHeaders()
     {
-        return httpThrottlingHeadersMapBuilder.build();
+        Map<String, String> throttlingHeaders = httpThrottlingHeadersMapBuilder.build();
+
+        for (String headerName : extraHeaders.keySet())
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Iterate over the "entrySet" instead of the "keySet". ([squid:S2864](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2864))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7625&issues=AWiOv-9iotIXnmsU1tFu&open=AWiOv-9iotIXnmsU1tFu)

[ISSUE_KEY]: AWiOv-9iotIXnmsU1tFu